docs(authorization.md): add authorization doc

vdice opened the pull request:
* Adds the Authorization doc

Admittedly it's fairly no-frills as written -- please advise if there are areas where more elaboration or context would be favored @krancour.

Closes https://github.com/brigadecore/brigade/issues/1575

radu-matei commented on docs/content/topics/administrators/authorization.md:
```diff
@@ -11,4 +11,117 @@ aliases:
+selected third-party auth provider and the creation of service accounts and
+role assignments is the responsibility of the administrator of Brigade.
+
+Note: There is one method of auto-creation of role assignments for a given set
```
nit: do you want to make this an actual markdown note?

krancour replied:
I think this is how we've generally done "notes" elsewhere, but now I'm curious...

How is a proper note done in markdown? Searching for this hasn't turned up anything especially useful.

radu-matei replied:
“>” at the beginning of the paragraph should make it a blockquote.

krancour replied:
Ah... I like the look of that, but always wondered if that wasn't a misuse of blockquotes. When I see that, I always assume I am looking at an actual quote.

vdice replied:
Thanks! Converted to a blockquote.

radu-matei commented on docs/content/topics/administrators/authorization.md:
```diff
@@ -11,4 +11,117 @@ aliases:
+assigned [roles](#roles) granting scoped permissions around their interactions
+with resources in Brigade.
+
+Administrators may list users, get a particular user's details, lock a user out
```
nit: “a user” repeated here makes this read a bit clunky.

radu-matei commented on docs/content/topics/administrators/authorization.md:
```diff
@@ -11,4 +11,117 @@ aliases:
+A Service Account in Brigade represents a non-human actor that can be assigned
+a [role](#roles) granting scoped permissions for interacting with resources in
+Brigade. A common pattern is to create a service account for a gateway and
+assign it an EVENT_CREATOR role such that it may submit events into Brigade.
```
nit: formatting for `EVENT_CREATOR`.

krancour commented:
This doc is very good. I have only one bit of feedback apart from @radu-matei's.

It might be worth mentioning that when a user with `PROJECT_CREATOR` creates a new project, they automatically receive `PROJECT_ADMIN`, `PROJECT_DEVELOPER`, and `PROJECT_USER` on that project.

vdice commented:
> It might be worth mentioning that when a user with PROJECT_CREATOR creates a new project, they automatically receive PROJECT_ADMIN, PROJECT_DEVELOPER, and PROJECT_USER on that project.

Good call; added.  Also added a note around the `source` requirement when assigning the EVENT_CREATOR role...

krancour reviewed (approved):
LGTM

vdice commented:
(Pushing an amended commit to re-trigger CI; no changes added.)
